lib/src/components/AbstractDialog/AbstractDialog.tsx

```diff
@@ -183,10 +189,29 @@ const AbstractDialog = forwardRef<HTMLDialogElement, Props>((props, ref) => {
       }
     };
 
+    // Prevent leak when focus goes out of the dialog modal.
+    const handleFocusIn = () => {
+      const dialogEl = dialogRef.current;
+      if (dialogEl && document.activeElement) {
+        const focusIsInsideDialogModal = document.activeElement.closest('dialog[aria-modal="true"][open]');
+        if (!focusIsInsideDialogModal) {
+          const firstFocusableElement = getFirstVisibleFocusableElement(dialogEl);
+          if (firstFocusableElement) {
+            firstFocusableElement.focus();
+          }
+        }
+      }
+    };
```

Comment:
Is there a scenario when the focus goes out of the dialog modal? I'm not entirely sure if this is required?

Reply:
> Is there a scenario when the focus goes out of the dialog modal? I'm not entirely sure if this is required?

There is, for example, when focused on the submit button on a form and submits, the submit button is removed and the focus moves out to body(or null) which moves it out of the modal

Reply:
> There is, for example, when focused on the submit button on a form and submits, but there is a validation error, the submit button is removed and the focus moves out to body(or null) which moves it out of the modal

I suspect it would be cleaner to implement this with `focusout`, using `relatedTarget` to detect whether the focus is leaving the modal or just moving around within it

Reply:
> > There is, for example, when focused on the submit button on a form and submits, but there is a validation error, the submit button is removed and the focus moves out to body(or null) which moves it out of the modal
> 
> I suspect it would be cleaner to implement this with `focusout`, using `relatedTarget` to detect whether the focus is leaving the modal or just moving around within it

Oo yeah that would be cleaner: a777480372daee2c407a4f1464e25901c40d741d